lib/vagrant-openshift/helper/command_helper.rb

```diff
@@ -160,7 +160,9 @@ def repo_checkout_bash_command(repo, url)
         git fetch origin
         git fetch --prune origin +refs/tags/*:refs/tags/*
         git reset --merge
-        git checkout #{branch}
+        if ! git checkout #{branch}; then
+          git checkout -b #{branch} origin/#{branch}
```

Comment:
When your origin is not your upstream, i.e. when someone's cloning Origin, their main remote will be openshift/origin, adding their own remote is optional. Not sure this code ever anticipated the main remote being someone's fork